Lib/test/test_urllib.py

```diff
@@ -426,6 +426,22 @@ def test_ftp_nonexisting(self):
         self.assertFalse(e.exception.filename)
         self.assertTrue(e.exception.reason)
 
+    def test_ftp_illegalhost(self):
+        illegal_ftp_hosts = [
+            'ftp://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'fTp://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'FTP://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'ftp://foo:bar%0aINJECTED@example.net/file.png',
+            'fTp://foo:bar%0aINJECTED@example.net/file.png',
+            'FTP://foo:bar%0aINJECTED@example.net/file.png'
+            ]
+
+        for host in illegal_ftp_hosts:
+            with self.assertRaises(urllib.error.URLError) as e:
+                urlopen(host)
+            self.assertFalse(e.exception.filename)
+            self.assertTrue(e.exception.reason)
```

Comment:
For example, the test pass if the connection fails with <urlopen error ftp error ConnectionRefusedError(111, 'Connection refused')>